addons/project_hr_expense/models/project.py

```diff
@@ -64,14 +69,20 @@ def _get_expenses_profitability_items(self, with_action=True):
         if not self.analytic_account_id:
             return {}
         can_see_expense = with_action and self.user_has_groups('hr_expense.group_hr_expense_team_approver')
-        expenses_read_group = self.env['hr.expense'].sudo()._read_group(
-            [('analytic_distribution_stored_char', '=ilike', f'%"{self.analytic_account_id.id}":%'),
-             ('is_refused', '=', False),
-             ('state', 'in', ['approved', 'done'])],
-            ['untaxed_amount', 'ids:array_agg(id)'],
-            [],
-        )
-        if not expenses_read_group or not expenses_read_group[0]['__count']:
+        query = self.env['hr.expense']._search([('is_refused', '=', False), ('state', 'in', ['approved', 'done'])])
+        query.add_where('hr_expense.analytic_distribution ? %s', [str(self.analytic_account_id.id)])
+        from_clause, where_clause, where_param = query.get_sql()
+        query = """
+            SELECT array_agg(id) as ids, SUM(untaxed_amount) as untaxed_amount
+            FROM %(from)s
+            WHERE %(where)s
+        """ % {
+            'from': from_clause,
+            'where': where_clause,
+        }
+        self._cr.execute(query, where_param)
```

Comment:
You should be able to do a `.select('array_agg(id) as ids', 'SUM(untaxed_amount) as untaxed_amount')` (only because there is no `GROUP BY` clause though)
```suggestion
        self.env.cr.execute(*query.select(
            'array_agg(id) as ids',
            'SUM(untaxed_amount) as untaxed_amount',
        ))